spec/webauthn/public_key_credential_spec.rb

```diff
@@ -135,11 +131,10 @@
 
       context "is received" do
         let(:attestation_response) do
-          allow_any_instance_of(WebAuthn::FakeAuthenticator::AuthenticatorData)
-            .to receive(:extensions)
-            .and_return({ "txAuthSimple" => "Could you please verify yourself?" })
-
-          response = client.create(challenge: raw_challenge)["response"]
+          response = client.create(
+            challenge: raw_challenge,
+            extensions: { "txAuthSimple" => "Could you please verify yourself?" }
+          )["response"]
```

Comment:
This is just a simple implementation of the code for faking the extensions. Right now, you require an extension when creating the request – either for authentication or registration – and that extension is returned with the same value from the request in both client and authenticator extension outputs.
I was thinking about doing something more realistic that processes each extension input and returns an output that is more similar to what each extension would return if you were using it in a real case based on the documentation, using some kind of `ExtensionProcesser`. I gave the idea some thinking but decided to keep it simple.

Reply:
> This is just a simple implementation of the code for faking the extensions. Right now, you require an extension when creating the request – either for authentication or registration – and that extension is returned with the same value from the request in both client and authenticator extension outputs.
> I was thinking about doing something more realistic that processes each extension input and returns an output that is more similar to what each extension would return if you were using it in a real case based on the documentation, using some kind of `ExtensionProcesser`. I gave the idea some thinking but decided to keep it simple.

This is an interesting/cool idea to consider.
For now I agree with you that we can keep it simple.

> 
> It's worth mention that before these changes, If you call the methods `make_credential` and `get_assertion` from `FakeAuthenticator` – the same goes for `create` and `get` from `FakeClient` which uses those methods – you have to specifically require for an extension in order to then receive an output in the authenticator data. That was not happening before, as we were not specifying the `extensions` key when initializing `AuthenticatorData`, so the default extension `{ "fakeExtension" => "fakeExtensionValue" }` was being used.
> IMO I like it better this way since no extension output is returned unless you specifically require one, but we can be potentially breaking some tests with this changes slightly_smiling_face

Makes sense, no extension output by default.

Thanks for the write up.